### PR TITLE
Clean up samplers

### PR DESCRIPTION
**Pull request by paolafer**
This PR addresses some points of issue #2, with a review of the samplers used in vertex generation.

- The `MuonsPointSampler` class is deleted, since the same sampling can be achieved with a box sampler.
- The class used currently for the cylindric sampler is renamed to `CylinderPointSampler` and the old one to `CylinderPointSamplerLegacy`.
- A new class for the box sampler is introduced, similar to the new one for the cylinder, following the G4 constructor of `G4Box` and with the possibility of passing a physical volume.
- New tests are added.
- `Enum` are used for regions, instead of `strings`.
- The `SpherePointSampler` class is also modified to adapt it to the G4 constructor.

### CODE REVIEW

File: source/utils/BoxPointSampler.h [hunk 1]
```diff
@@ -9,28 +9,32 @@
...
+namespace nexus {
+  
   class BoxPointSampler
   {
   public:
     /// Constructor
-    BoxPointSampler(G4double inner_x, G4double inner_y, G4double inner_z,
-                    G4double thickness,
+    BoxPointSampler(G4double half_inner_x, G4double half_inner_y,
+                    G4double half_inner_z, G4double thickness=0,
```
Comment (atrettin) [inline source/utils/BoxPointSampler.h:27]: Just out of interest: Why is this "half" everywhere now? I see many factors of two or half have been added in the code to get the same result.

Comment (paolafer) [reply source/utils/BoxPointSampler.h:27]: The guide line here is to mimic the constructor of the correspondent solid volumes, in this case, `G4Box` (see, for instance, https://geant4.kek.jp/lxr/source/geometry/solids/CSG/include/G4Box.hh#L59).

Comment (atrettin) [reply source/utils/BoxPointSampler.h:27]: I see. I'm not a huge fan of having all these divisions by two everywhere in the code now, but I suppose that makes sense if we want to match our convention to that of G4. Thank you for the explanation.

File: source/geometries/Next1EL.cc [hunk 2]
```diff
@@ -1371,7 +1373,7 @@ G4ThreeVector Next1EL::GenerateVertex(const G4String& region) const
       }
   } else if (region == "MUONS") {
     //generate muons sampling the plane
-    vertex = muons_sampling_->GenerateVertex();
+    vertex = muons_sampling_->GenerateVertex("INSIDE");
```
Comment (atrettin) [inline source/geometries/Next1EL.cc:1376]: Why doesn't this use an enum?

Comment (atrettin) [reply source/geometries/Next1EL.cc:1376]: Nevermind, I see that this is calling the legacy sampler where it is indeed a string.

**Review by atrettin — APPROVED** [review]
I have compiled and run `nexus` with these changes and looked at all the files. The re-naming of the classes and change of conventions have been done consistently as far as I can see.

**Pull request merged**